api/src/main/java/org/openmrs/scheduler/tasks/AutoRetireUsersTask.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.scheduler.tasks;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.User;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.util.DateUtil;
+import org.openmrs.util.OpenmrsConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.openmrs.util.OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS;
+
+/**
+ * A scheduled task that automatically retires user after the set number-of-days of inactivity. 
+ * The inactivity duration is set as a global property 
+ * Documentation: https://openmrs.atlassian.net/wiki/spaces/docs/pages/25520634/Administering+Scheduled+Tasks
+ * {@link org.openmrs.util.OpenmrsConstants#GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS}
+ */
+public class AutoRetireUsersTask extends AbstractTask {
+	
+	private static final Logger log = LoggerFactory.getLogger(AutoRetireUsersTask.class);
+
+	/**
+	 * @see org.openmrs.scheduler.tasks.AbstractTask#execute()
+	 */
+	@Override
+	public void execute() {
+		if (!isExecuting) {
+			log.debug("Starting auto-retiring users");
+			
+			startExecuting();
+			
+			try {
+				UserService userService = Context.getUserService();
+				Set<User> usersToRetire = getUsersToRetire(userService);
+				
+				usersToRetire.forEach(user -> userService.retireUser(user, "Retire reason"));
+			} catch (Exception e) {
+				log.error("Error occurred while auto-retiring users: ", e);
+			} finally {
+				log.debug("Auto-retiring users task ended");
+				stopExecuting();
+			}
+		}
+	}
+
+	private Set<User> getUsersToRetire(UserService userService) {
+		List<User> allUsers = userService.getAllUsers();
+		String numberOfDaysToRetire = Context.getAdministrationService().getGlobalProperty(GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS);
```

Comment:
What happens when this global property is not yet set or has an invalid (non numeric) value?

Reply:
I have added a null check incase the value is not set. Tested to be working well.

Reply:
How about if it is simply an empty string?